cmd/panfs.go

```diff
@@ -775,8 +775,19 @@ func (fs *PANFSObjects) DeleteBucket(ctx context.Context, bucket string, opts De
 		return toObjectErr(err, bucket)
 	}
 
-	// Delete all bucket metadata.
-	deleteBucketMetadata(ctx, fs, bucket)
+	if fs.configAgent != nil {
+		noLockID := ""
+		if err = fs.configAgent.DeleteObject(
+			pathJoin(panfsBucketListPrefix, bucket), noLockID); err != nil {
+			return toObjectErr(err, bucket)
+		}
+	} else {
+		if err = fsRemoveAll(ctx, pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, bucket)); err != nil {
```

Comment:
The problem you are solving here by adding the `fsRemoveAll()` call is not related to the fact that the config agent is used. If that's OK with you, let's leave the `if fs.configAgent != nil {` part where it originally was and just make sure `fsRemoveAll()` is called before `deleteBucketMetadata()`.